LayoutTests/inspector/runtime/callFunctionOn-awaitPromise.html

```diff
@@ -0,0 +1,141 @@
+<script>
+if (window.internals)
+    window.internals.settings.setUnhandledPromiseRejectionToConsoleEnabled(false);
+
+const myObject = {a: 2023};
+
+function test()
+{
+    const expectedObject = {a: 2023};
+
+    const suite = InspectorTest.createAsyncSuite("Runtime.callFunction.awaitPromise");
+
+    suite.addTestCase({
+        name: "NoAwaitSyncFunction",
+        description: "Check that callFunctionOn with awaitPromise=false works with a synchronous function.",
+        async test() {
+            const thisObject = await RuntimeAgent.evaluate.invoke({expression: `myObject`});
+            const result = await RuntimeAgent.callFunctionOn.invoke({functionDeclaration: `function() { return this; }`, awaitPromise: false, returnByValue: true, objectId: thisObject.result.objectId});
+            InspectorTest.expectShallowEqual(result.result.value, expectedObject, `Should return the JSON object.`);
+        },
+    });
+
+    suite.addTestCase({
+        name: "NoAwaitSyncFunctionException",
+        description: "Check that callFunctionOn with awaitPromise=false works with a synchronous function.",
+        async test() {
+            const thisObject = await RuntimeAgent.evaluate.invoke({expression: `myObject`});
+            const result = await RuntimeAgent.callFunctionOn.invoke({functionDeclaration: `function() { throw "My error"; }`, awaitPromise: false, returnByValue: true, objectId: thisObject.result.objectId});
+            InspectorTest.expectThat(result.wasThrown, "Should throw an error.");
+            InspectorTest.expectEqual(result.result.value, "My error", `Should return the error.`);
+        },
+    });
+
+    suite.addTestCase({
+        name: "NoAwaitAsyncFunction",
+        description: "Check that callFunctionOn with awaitPromise=false works with an asynchronous function.",
+        async test() {
+            const thisObject = await RuntimeAgent.evaluate.invoke({expression: `myObject`});
+            const result = await RuntimeAgent.callFunctionOn.invoke({functionDeclaration: `async function() { await new Promise((fulfill, reject) => setTimeout(fulfill, 10)); return this; }`, awaitPromise: false, returnByValue: true, objectId: thisObject.result.objectId});
+            InspectorTest.expectShallowEqual(result.result.value, {}, `Should return empty object.`);
+        },
+    });
+
+    suite.addTestCase({
+        name: "NoAwaitPromiseRejection",
+        description: "Check that callFunctionOn with awaitPromise=false works with an asynchronous function.",
+        async test() {
+            const thisObject = await RuntimeAgent.evaluate.invoke({expression: `myObject`});
+            const result = await RuntimeAgent.callFunctionOn.invoke({functionDeclaration: `async function() { return Promise.reject(this); }`, awaitPromise: false, returnByValue: true, objectId: thisObject.result.objectId});
+            InspectorTest.expectThat(!result.wasThrown, "Should not throw an error.");
```

Comment:
NIT: id use `InspectorText.expectFalse` instead for this (and `InspectorText.expectTrue` for the other cases like this)